src/connection/mod.rs

```diff
@@ -178,114 +203,53 @@ fn test_enable_disable_functions() {
-                                    dbus::arg::Dict<&str, dbus::arg::Variant<dbus::arg::Iter>, _>,
-                                    _> = response.get1().unwrap();
-
-    for (_, v1) in dictionary {
-        for (k2, v2) in v1 {
-            match k2 {
-                "id" => {
-                    connection.id = v2.0
-                        .clone()
-                        .get::<&str>()
-                        .unwrap()
-                        .to_string();
-                }
-                "uuid" => {
-                    connection.uuid = v2.0
-                        .clone()
-                        .get::<&str>()
-                        .unwrap()
-                        .to_string();
-                }
-                "ssid" => {
-                    connection.ssid =
-                        std::str::from_utf8(&v2.0
-                                                 .clone()
-                                                 .get::<dbus::arg::Array<u8, _>>()
-                                                 .unwrap()
-                                                 .collect::<Vec<u8>>())
-                                .unwrap()
-                                .to_string();
-                }
-                _ => (),
-            }
-        }
-    }
+    connection.path = path.clone();
+
+    connection.settings = try!(manager.get_connection_settings(path));
 
-    update_state(&mut connection).unwrap();
+    try!(update_state(manager, &mut connection));
 
     Ok(connection)
 }
 
-fn update_state(connection: &mut Connection) -> Result<(), String> {
-    let active_paths: Vec<String> = dbus_property!(NM_SERVICE_MANAGER,
-                                                   NM_SERVICE_PATH,
-                                                   NM_SERVICE_INTERFACE,
-                                                   "ActiveConnections")
-            .unwrap()
-            .inner::<&Vec<dbus::MessageItem>>()
-            .unwrap()
-            .iter()
-            .map(|p| dbus_path_to_string(p.inner::<&dbus::Path>().unwrap().to_owned()))
-            .collect();
-
-    let settings_paths = active_paths.iter().map(|p| {
-        dbus_path_to_string(dbus_property!(NM_SERVICE_MANAGER,
-                                           p,
-                                           NM_ACTIVE_INTERFACE,
-                                           "Connection")
-                                    .unwrap()
-                                    .inner::<&dbus::Path>()
-                                    .unwrap()
-                                    .to_owned())
-    });
+fn update_state(manager: &NetworkManager, connection: &mut Connection) -> Result<(), String> {
+    let active_paths = try!(manager.get_active_connections());
+
+    let mut settings_paths = Vec::new();
 
+    for active_path in &active_paths {
+        if let Some(settings_path) = manager.get_active_connection_path(&active_path) {
```

Comment:
Does this fix the failing test issue where the active connection no longer exists?